### PR TITLE
[DM-1294]: Added 3 announcements for device grid changes.

### PR DESCRIPTION
**Pull request by dafkk**
One in 1017 about the deprecation of the data grid logic, the same in 1018, and one in 1019 about the removal of the logic.

### CODE REVIEW

File: content/release-10-17-0/announcements-10-17-0.md [hunk 1]
```diff
@@ -168,6 +168,12 @@ The HOOK_SEARCH can be used by developers to extend the web integration of the s
 
 As announced with [release 10.16](/release-10-16-0/announcements-10-16-0), with release 10.18+, the "Map" widget will be migrated and support for real-time updates on all devices will be replaced by a configurable refresh interval. This change allows us to display more then 100 devices on the map by default. Additionally, the "Map" widget will no longer support the "show track" option.
 
+##### Deprecated device-grid model classes, column implementations and services
+
+Shared classes, components and services from the device-grid package are deprecated and new ones are created in the core data-grid component.
+
+This change only affects you, if you or your development team use the Web SDK to extend Cumulocity IoT UI applications or to build your own web applications. If you do and you do use the device-grid functionalities, check the deprecation documentation and alter your code accordingly. Refer to the deprecations in the WebSDK resources documentation for the device-grid service [here](http://resources.cumulocity.com/documentation/websdk/ngx-components/injectables/DeviceGridService.html). Other deprecations for reference are also marked in this documentation.
```
Comment (BeateRixen) [inline content/release-10-17-0/announcements-10-17-0.md:175]:
```suggestion
This change only affects you, if you or your development team use the Web SDK to extend Cumulocity IoT UI applications or to build your own web applications. If you use the device-grid functionalities, check the deprecation documentation and alter your code accordingly. Refer to the deprecations in the [WebSDK resources documentation for the device-grid service](http://resources.cumulocity.com/documentation/websdk/ngx-components/injectables/DeviceGridService.html). Other deprecations for reference are also marked in this documentation.
```

File: content/release-10-17-0/announcements-10-17-0.md [hunk 2]
```diff
@@ -168,6 +168,12 @@ The HOOK_SEARCH can be used by developers to extend the web integration of the s
 
 As announced with [release 10.16](/release-10-16-0/announcements-10-16-0), with release 10.18+, the "Map" widget will be migrated and support for real-time updates on all devices will be replaced by a configurable refresh interval. This change allows us to display more then 100 devices on the map by default. Additionally, the "Map" widget will no longer support the "show track" option.
 
+##### Deprecated device-grid model classes, column implementations and services
```
Comment (BeateRixen) [inline content/release-10-17-0/announcements-10-17-0.md:171]:
```suggestion
##### Deprecation of device-grid model classes, column implementations and services
```

File: content/release-2024-0/announcements-2024-0.md [hunk 3]
```diff
@@ -26,6 +26,11 @@ This change affects the following APIs:
 
 #### Implemented
 
+##### Removal of deprecated device-grid model classes, column implementations and services
+
+As announced with [release 10.17](/release-10-17-0/announcements-10-17-0) shared classes, components and services from the device-grid package are deprecated and new ones are created in the core data-grid component. All those deprecated classes are now removed.
```
Comment (BeateRixen) [inline content/release-2024-0/announcements-2024-0.md:31]:
```suggestion
As announced earlier, most recently with [release 10.17](/release-10-17-0/announcements-10-17-0), shared classes, components and services from the device-grid package are deprecated and have now been removed. New ones are created in the core data-grid component.
```

File: content/release-10-18-0/announcements-10-18-0.md [hunk 4]
```diff
@@ -144,13 +144,18 @@ All of the impacted injection tokens have been marked as deprecated with 10.18,
 
 The HOOK_SEARCH can be used by developers to extend the web integration of the search in a custom web application. As announced with [release 10.17](/release-10-17-0/announcements-10-17-0), as of a future release, we will refactor the HOOK_SEARCH interface. If you use this interface in a custom-developed UI application, you must migrate to the new version. Details on how to use the new interface will be provided in the Web SDK documentation of the respective release.
 
+##### Deprecated device-grid model classes, column implementations and services
+
+As announced with [release 10.17](/release-10-17-0/announcements-10-17-0) shared classes, components and services from the device-grid package are deprecated and new ones are created in the core data-grid component.
+
+This change only affects you, if you or your development team use the Web SDK to extend Cumulocity IoT UI applications or to build your own web applications.
```
Comment (BeateRixen) [inline content/release-10-18-0/announcements-10-18-0.md:151]:
```suggestion
This change only affects you, if you or your development team use the Web SDK to extend Cumulocity IoT UI applications or to build your own web applications. If you use the device-grid functionalities, check the deprecation documentation and alter your code accordingly. Refer to the deprecations in the [WebSDK resources documentation for the device-grid service](http://resources.cumulocity.com/documentation/websdk/ngx-components/injectables/DeviceGridService.html). Other deprecations for reference are also marked in this documentation.
```

**Review by BeateRixen — APPROVED** [review]

File: content/release-10-18-0/announcements-10-18-0.md [hunk 5]
```diff
@@ -144,13 +144,18 @@ All of the impacted injection tokens have been marked as deprecated with 10.18,
 
 The HOOK_SEARCH can be used by developers to extend the web integration of the search in a custom web application. As announced with [release 10.17](/release-10-17-0/announcements-10-17-0), as of a future release, we will refactor the HOOK_SEARCH interface. If you use this interface in a custom-developed UI application, you must migrate to the new version. Details on how to use the new interface will be provided in the Web SDK documentation of the respective release.
 
+##### Deprecation of device-grid model classes, column implementations and services
+
+As announced with [release 10.17](/release-10-17-0/announcements-10-17-0) shared classes, components and services from the device-grid package are deprecated and new ones are created in the core data-grid component.
```
Comment (bozhidardedov) [inline content/release-10-18-0/announcements-10-18-0.md:149]: Maybe rather mention that common grid API components will be **moved** to `@c8y/ngx-components`, therefore the need for obsolete classes, components and services in `@c8y/ngx-components/device-grid` to be deprecated.

**Review by bozhidardedov — APPROVED** [review]

File: content/release-10-18-0/announcements-10-18-0.md (hunk 5 above)
Comment (dafkk) [reply content/release-10-18-0/announcements-10-18-0.md]: Updated, you can take a look again.

File: content/release-10-17-0/announcements-10-17-0.md [hunk 6]
```diff
@@ -168,6 +168,12 @@ The HOOK_SEARCH can be used by developers to extend the web integration of the s
 
 As announced with [release 10.16](/release-10-16-0/announcements-10-16-0), with release 10.18+, the "Map" widget will be migrated and support for real-time updates on all devices will be replaced by a configurable refresh interval. This change allows us to display more then 100 devices on the map by default. Additionally, the "Map" widget will no longer support the "show track" option.
 
+##### Deprecation of device-grid model classes, column implementations and services
+
+Shared classes, components and services defined in @c8y/ngx-components/device-grid are now moved to @c8y/ngx-components. Therefore the obsolete items are deprecated.
```
Comment (BeateRixen) [inline content/release-10-17-0/announcements-10-17-0.md:173]:
```suggestion
Shared classes, components and services previously defined in @c8y/ngx-components/device-grid have now been moved to @c8y/ngx-components. Therefore the obsolete items are deprecated.
```

File: content/release-10-18-0/announcements-10-18-0.md [hunk 7]
```diff
@@ -144,13 +144,18 @@ All of the impacted injection tokens have been marked as deprecated with 10.18,
 
 The HOOK_SEARCH can be used by developers to extend the web integration of the search in a custom web application. As announced with [release 10.17](/release-10-17-0/announcements-10-17-0), as of a future release, we will refactor the HOOK_SEARCH interface. If you use this interface in a custom-developed UI application, you must migrate to the new version. Details on how to use the new interface will be provided in the Web SDK documentation of the respective release.
 
+##### Deprecation of device-grid model classes, column implementations and services
+
+As announced with [release 10.17](/release-10-17-0/announcements-10-17-0) shared classes, components and services defined in @c8y/ngx-components/device-grid are now moved to @c8y/ngx-components. Therefore the obsolete items are deprecated.
```
Comment (BeateRixen) [inline content/release-10-18-0/announcements-10-18-0.md:149]:
```suggestion
As announced with [release 10.17](/release-10-17-0/announcements-10-17-0) shared classes, components and services previously defined in @c8y/ngx-components/device-grid have been moved to @c8y/ngx-components. Therefore the obsolete items are deprecated.
```

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**